platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java

```diff
@@ -603,17 +610,21 @@ public VirtualLeafRecord<K, V> loadLeafRecord(final K key) throws IOException {
     }
 
     /**
-     * Load a leaf record by path
+     * Load a leaf record by path. This method returns {@code null}, if the path is outside the
+     * valid path range.
      *
      * @param path the path for the leaf we are loading
      * @return loaded record or null if not found
      * @throws IOException If there was a problem reading record from db
      */
     @Override
     public VirtualLeafRecord<K, V> loadLeafRecord(final long path) throws IOException {
+        if (path < 0) {
+            throw new IllegalArgumentException("Path (" + path + ") is not valid");
+        }
         final KeyRange leafPathRange = validLeafPathRange;
         if (!leafPathRange.withinRange(path)) {
-            throw new IllegalArgumentException("path (" + path + ") is not valid; must be in range " + leafPathRange);
+            return null;
```

Comment:
This behavior is changed for consistency with related methods like `loadHash()`, which return null (rather than throw an exception), if a path is outside the current path range